CMakeLists.txt

```diff
@@ -83,7 +83,7 @@ elseif(EMSCRIPTEN)
     # set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${EM_FLAGS}")
 else()
     add_compile_definitions(__linux__=1)
-    set(CMAKE_CXX_FLAGS "-w -std=c++11 -DUNIX -pthread")
+    set(CMAKE_CXX_FLAGS "-w -std=c++17 -DUNIX -pthread")
```

Comment:
> [Eigen has good compiler support](https://eigen.tuxfamily.org/index.php?title=Main_Page#Compiler_support) as we run our test suite against many compilers to guarantee reliability and work around any compiler bugs. Eigen up to version 3.4 is standard C++03 and maintains reasonable compilation times. Versions following 3.4 will be C++14.

Eigen requires C++14 for now strictly speaking. But C++17 would be my preferred standard to target, I don't know if it will create side effects on our side though, but if CI passes here I'm fine with it

Reply:
Thanks for the response, please have a double check. 
Best
Chi